tests/e2e/util.go

```diff
@@ -941,6 +941,31 @@ func updateCSIDeploymentTemplateFullSyncInterval(client clientset.Interface, min
 	return oldValue
 }
 
+// updateCSIDeploymentProvisionerTimeout helps to update the timeout value
+// in deployment template. For this to take effect,
+func updateCSIDeploymentProvisionerTimeout(client clientset.Interface, namespace string, timeout string) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	deployment, err := client.AppsV1().Deployments(namespace).Get(
+		ctx, vSphereCSIControllerPodNamePrefix, metav1.GetOptions{})
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	containers := deployment.Spec.Template.Spec.Containers
+	for _, c := range containers {
+		if c.Name == "csi-provisioner" {
+			for i, e := range c.Args {
+				if strings.Contains(e, "timeout") {
+					c.Args[i] = "--timeout=" + timeout + "s"
+				}
+			}
+		}
+	}
+	deployment.Spec.Template.Spec.Containers = containers
+	_, err = client.AppsV1().Deployments(namespace).Update(ctx, deployment, metav1.UpdateOptions{})
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	ginkgo.By("Waiting for a min for update operation on deployment to take effect...")
```

Comment:
rather than this wait for the csi pod to be up and running

Reply:
Need a wait here because, if we check the status of CSI pods immediately after the update - all the pods will be in running state - so it does not wait for the CSI controller pod to terminate and be running again.

I will add an additional check to see if all the pods are in running state after one min.